scripts/build_dependencies/build_utils.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+const child_process = require("child_process");
+
+module.exports = {
+
+    npmCheckIfPackageExists: function (package_name, package_version) {
+
+        // Do we have it in our node list? If so, then use that!
+        try {
+            var list_output = child_process.execSync("npm list --depth 0 " + package_name, {encoding: "utf8"});
+            if (list_output.indexOf(package_name + "@" + package_version) !== -1) {
+                console.log("Found " + package_name + " in node list!");
+                return true;
+            }
+        } catch (error) {}
+
+        // Do we have it in our global list?
+        try {
+            var list_output = child_process.execSync("npm list -g --depth 0 " + package_name, {encoding: "utf8"});
+            if (list_output.indexOf(package_name + "@" + package_version) !== -1) {
+                console.log("Found " + package_name + " in node list!");
+                return true;
+            }
+        } catch (error) {}
```

Comment:
I'm not sure if there's any effective difference to what you have here, but you can use `require.resolve('package_name')` to check for existence and `require('package_name/package.json').version` to check the version.

Since you will be calling `require` for the package(s), `require.resolve` uses the same lookup.

Reply:
I looked at using `require.resolve` and while it works for the check, then trying to use `require` after installing the package leads to a `elifecycle errno 1` crash. It seems that something about using `require.resolve` makes a following `require` call in the same Javascript instance not work. It works fine if you then rerun the script again though, but it doesn't allow for checking, seeing it is not there, downloading the package, and then using it all in one go.

I tried unloading after the `require.resolve` call, but this didn't work either. For now, I'm going to use the current solution since it seems to work but I'll add a TODO note to look at changing this to `require.resolve` in the future.